src/whisper-flat.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#ifdef WHISPER_SHARED
+#    ifdef _WIN32
+#        ifdef WHISPER_BUILD
+#            define WHISPER_FLAT_API __declspec(dllexport)
+#        else
+#            define WHISPER_FLAT_API __declspec(dllimport)
+#        endif
+#    else
+#        define WHISPER_FLAT_API __attribute__ ((visibility ("default")))
+#    endif
+#else
+#    define WHISPER_FLAT_API
+#endif
+
+#include "whisper.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+    #ifdef BINDINGS_FLAT
+    WHISPER_FLAT_API void whisper_flat_backend_load_all(void);
+    WHISPER_FLAT_API struct whisper_activity * whisper_flat_get_activity_with_state(struct whisper_state * state);
+    WHISPER_FLAT_API struct whisper_state * whisper_flat_get_state_from_context(struct whisper_context * ctx);
```

Comment:
Same question as above, this should really be exposed?

Reply:
They exist precisely owing to the fact that c++ std:vector aren't available in other languages
Externally whisper_state can be treated as an opaque object as long as you have its address so whisper_flat_get_state_from_context can pluck it out of a whisper_context then plug it into whisper_flat_get_activity_with_state to get timing data directly rather than via JSON